engine/src/execution_graph/logic_controllers/CacheMachine.cpp

```diff
@@ -68,7 +70,22 @@ CacheMachine::~CacheMachine() {}
 void CacheMachine::finish() {
 	this->waitingCache->finish();
 }
+
+bool CacheMachine::is_finished() {
+	return this->waitingCache->is_finished();
+}
+
+uint64_t CacheMachine::get_num_bytes_added(){
+	return num_bytes_added.load();
+}
+
+uint64_t CacheMachine::get_num_rows_added(){
+	return num_rows_added.load();
+}
+
 void CacheMachine::addHostFrameToCache(std::unique_ptr<ral::frame::BlazingHostTable> host_table, std::string message_id) {
```

Comment:
there should only be one add to cache and it adds a blazingtable the output from a kernel, its when it gets added to the cache that it gets coverted to different kind of frames, why are there different apis for this?

Reply:
There are three APIs. Once for what you describe. Another to add HostFrames which is what comes in via a shuffle. We dont want to bring that into the GPU, we want to leave that in RAM until it gets explicity decached when exiting the cache. Third API is to pass a cache object from one cache to another is a passthrough sort of way, in that case, you dont want to change the state of the cache object

Reply:
ok